hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateMap.java

```diff
@@ -70,14 +70,6 @@ class PipelineStateMap {
    * @throws IOException if pipeline with provided pipelineID already exists
    */
   void addPipeline(Pipeline pipeline) throws IOException {
-    Preconditions.checkNotNull(pipeline, "Pipeline cannot be null");
```

Comment:
Do you think we should still have a check here, and at least log if the pipeline does not have enough nodes and skip it, rather than loading it? I guess we should not get a "bad pipeline" here, as it never should be allowed to be added anyway.

Reply:
I think it make sense to leave this preconditions as is. Anyway we are making sure to pass the correct pipeline with other checks.